src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -142,6 +142,112 @@ public static unsafe bool GetKnownMethod(this Span<byte> span, out HttpMethod me
             }
         }
 
+        /// <summary>
+        /// Parses string <paramref name="value"/> for a known HTTP method.
+        /// </summary>
+        /// <remarks>
+        /// A "known HTTP method" can be an HTTP method name defined in the HTTP/1.1 RFC.
+        /// The Known Methods (CONNECT, DELETE, GET, HEAD, PATCH, POST, PUT, OPTIONS, TRACE)
+        /// </remarks>
+        /// <returns><c>true</c> if the input matches a known string, <c>false</c> otherwise.</returns>
+        public static bool GetKnownMethod(string value, out HttpMethod method)
+        {
+            // Only called if user code overrides and sets the Method on IHttpRequestFeature
+            // This should be a rare occurance so not performance sensitive also needs to validate inputs
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            var length = value.Length;
+            if (length == 0)
+            {
+                throw new ArgumentException(nameof(value));
+            }
+
+            // Clear the 6th bit to get Upper case char. We are testing direct equality 
+            // so doesn't matter that it would garble chars outside [A-Za-z] range
+            var firstUppercaseChar = (char)(value[0] & ~0x20);
+            if (length == 3)
+            {
+                if (firstUppercaseChar == 'G' && string.Equals(value, "GET", StringComparison.OrdinalIgnoreCase))
```

Comment:
> `OrdinalIgnoreCase`

https://tools.ietf.org/html/rfc7230#section-3.1.1:

> The request method is case-sensitive.

Reply:
Somethings wonder if they flip coins to determine what is case sensitive in the specs. Will change

Reply:
When it doubt, use case in-sensitive for parsing, regardless of the spec.

Reply:
@Tratcher I would go with stricter parsing when in doubt. The risk is that Kestrel might interpret a request differently than a proxy. For instance, imagine a proxy is configured with rules that should be applied to all POST requests to a given endpoint. With case-insensitive parsing in Kestrel, these rules could possibly be bypassed by sending a pOsT request.